axopy/timing.py

```diff
@@ -78,6 +78,89 @@ def reset(self):
         self.count = 0
 
 
+class StepCounter(Counter):
+    """Multiple step counter.
+       Counts to a given number then transmits a timeout event.
+       Transmits events at multiple values up to timeout.
+
+    Parameters
+    ----------
+    max_count : int
+        Number of iterations to go through before transmitting the `timeout`
+        event. Must be greater than 1.
+    reset_on_timeout : bool, optional
+        Specifies whether or not the timer should reset its count back to zero
+        once the timeout event occurs. The default behavior is to reset.
+
+    Attributes
+    ----------
+    count : int
+        Current count.
+    timeout : Transmitter
+        Transmitted when ``max_count`` has been reached.
+
+    Examples
+    --------
+    Basic usage:
+
+    >>> from axopy.timing import StepCounter
+    >>> timer = StepCounter(3)
+    >>> timer.add_step(1, function1)
+    >>> timer.add_step(2, function2)
+    >>> timer.increment()
+    "function 1"
+    >>> timer.increment()
+    "function 2"
+    >>> timer.increment()
+    >>> timer.count
+    0
+    """
+
+    timeout = Transmitter()
+
+    # README: No way to create an array of pyqtSignal(s) ...
+    # https://stackoverflow.com/questions/38506979/creating-an-array-of-pyqtsignal 
+    step_max = 10
+    for i in range(step_max):
+        vars()['step' + str(i)] = Transmitter()
+
+    def __init__(self, max_count=1, reset_on_timeout=True):
+        super(StepCounter, self).__init__(max_count, reset_on_timeout)
+        self.step_inc = 0
+        self.step_count = []
+
+    def _dummy():
+        pass
```

Comment:
I understand that you define this so that you can have a default for `add_step()` below.  If so, just add a comment explaining that and make this a `@staticmethod`  as it doesn't use the instance. Alterantively, you can omit the default below and remove this altogether and expect the user to pass in a valid callback .